### PR TITLE
fix: Set correct workspace.version upon loading a workspace

### LINKED ISSUE #801: Workspace.version is not set when loading in a workspace with a version
# Description

See the title.

# Expected Behavior

`Workspace.version is not None`

# Actual Behavior

`Workspace.version is None`

# Steps to Reproduce

```python
import pyhf, json

ws = pyhf.Workspace(json.load(open('workspace.json')))
assert ws.version is not None
assert ws.version == ws['version']
```

# Checklist

- [x] Run `git fetch` to get the most up to date version of `master`
- [x] Searched through existing Issues to confirm this is not a duplicate issue
- [x] Filled out the Description, Expected Behavior, Actual Behavior, and Steps to Reproduce sections above or have edited/removed them in a way that fully describes the issue

### PR DESCRIPTION
**Pull request by kanishk16**
# Description

This PR fixes the bug `Workspace.version is not None` upon loading a workspace with a version by assigning the correct version to the instance variable `version` accordingly as per the arguments.

Resolves #801 

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR

```
* Set the workspace version while loading a workspace
* Add test to check loaded workspace for version
```

### CODE REVIEW

File: src/pyhf/workspace.py [hunk 1]
```diff
@@ -267,7 +267,12 @@ def __init__(self, spec, **config_kwargs):
         """Workspaces hold the model, data and measurements."""
         super(Workspace, self).__init__(spec, channels=spec['channels'])
         self.schema = config_kwargs.pop('schema', 'workspace.json')
-        self.version = config_kwargs.pop('version', None)
+
+        if 'version' in spec:
+            self.version = spec['version']
+        else:
+            self.version = config_kwargs.pop('version', None)
+
```
Comment (kratsg) [inline src/pyhf/workspace.py]: this should be done after schema validation which guarantees a `version` in the spec.

File: src/pyhf/workspace.py [hunk 2]
```diff
@@ -267,10 +267,15 @@ def __init__(self, spec, **config_kwargs):
...
         self.schema = config_kwargs.pop('schema', 'workspace.json')
-        self.version = config_kwargs.pop('version', None)
+
         # run jsonschema validation of input specification against the (provided) schema
-        log.info("Validating spec against schema: {0:s}".format(self.schema))
-        utils.validate(self, self.schema, version=self.version)
+        log.info(f"Validating spec against schema: {self.schema}")
+        utils.validate(self, self.schema)
+
+        if 'version' in spec:
```
Comment (kratsg) [inline src/pyhf/workspace.py]: this check is not needed since `version` must be in `spec` to pass schema validation.

Comment (kanishk16) [reply src/pyhf/workspace.py]: Yes, my bad. I'll make the changes.

**Comment by kratsg** [comment]
I'm realizing we're going to make this harder in the future unfortunately, this way. Let's move it back to before the schema validation and have the one-liner be

`self.version = config_kwargs.pop('version', spec.get('version', None))`

**Comment by kanishk16** [comment]
> I'm realizing we're going to make this harder in the future unfortunately, this way.
Would it be reasonable to ask for the intuition or the logic behind it?

**Comment by kratsg** [comment]
> Would it be reasonable to ask for the intuition or the logic behind it?

we need to know the version of the workspace to validate against the right version of the schema.

**Comment by kanishk16** [comment]
> Need a test to check for regression in `tests/test_workspace.py`

Now on to it.

**Pull request merged**